src/otx/core/model/entity/utils/mmaction.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Utility files for the mmaction package."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from mmaction.models.data_preprocessors import (
+    ActionDataPreprocessor as _ActionDataPreprocessor,
+)
+from mmaction.registry import MODELS
+from mmengine.registry import MODELS as MMENGINE_MODELS
+
+from otx.core.utils.build import build_mm_model, get_classification_layers
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+    from torch import device, nn
+
+
+# NOTE: For the history of this monkey patching, please see
+# https://github.com/openvinotoolkit/training_extensions/issues/2743
+@MMENGINE_MODELS.register_module(force=True)
+class ActionDataPreprocessor(_ActionDataPreprocessor):
+    """Class for monkey patching preprocessor.
+
+    NOTE: For the history of this monkey patching, please see
+    https://github.com/openvinotoolkit/training_extensions/issues/2743
+    """
+
+    @property
+    def device(self) -> device:
+        """Which device being used."""
+        try:
+            buf = next(self.buffers())
+        except StopIteration:
+            return super().device
+        else:
+            return buf.device
+
+
+def create_model(config: DictConfig, load_from: str | None) -> tuple[nn.Module, dict[str, dict[str, int]]]:
```

Comment:
The reason these functions can't be combined into one is because of mmX's import structure, right?

Reply:
you're right. Each framework needs to use own `MODELS` registry.